src/core/tests/float8_e8m0.cpp

```diff
@@ -0,0 +1,378 @@
+    3.6893488147419103e+19f,    7.378697629483821e+19f,     1.4757395258967641e+20f,    2.9514790517935283e+20f,
+    5.902958103587057e+20f,     1.1805916207174113e+21f,    2.3611832414348226e+21f,    4.722366482869645e+21f,
+    9.44473296573929e+21f,      1.888946593147858e+22f,     3.777893186295716e+22f,     7.555786372591432e+22f,
+    1.5111572745182865e+23f,    3.022314549036573e+23f,     6.044629098073146e+23f,     1.2089258196146292e+24f,
+    2.4178516392292583e+24f,    4.835703278458517e+24f,     9.671406556917033e+24f,     1.9342813113834067e+25f,
+    3.8685626227668134e+25f,    7.737125245533627e+25f,     1.5474250491067253e+26f,    3.094850098213451e+26f,
+    6.189700196426902e+26f,     1.2379400392853803e+27f,    2.4758800785707605e+27f,    4.951760157141521e+27f,
+    9.903520314283042e+27f,     1.9807040628566084e+28f,    3.961408125713217e+28f,     7.922816251426434e+28f,
+    1.5845632502852868e+29f,    3.1691265005705735e+29f,    6.338253001141147e+29f,     1.2676506002282294e+30f,
+    2.535301200456459e+30f,     5.070602400912918e+30f,     1.0141204801825835e+31f,    2.028240960365167e+31f,
+    4.056481920730334e+31f,     8.112963841460668e+31f,     1.6225927682921336e+32f,    3.2451855365842673e+32f,
+    6.490371073168535e+32f,     1.298074214633707e+33f,     2.596148429267414e+33f,     5.192296858534828e+33f,
+    1.0384593717069655e+34f,    2.076918743413931e+34f,     4.153837486827862e+34f,     8.307674973655724e+34f,
+    1.661534994731145e+35f,     3.32306998946229e+35f,      6.64613997892458e+35f,      1.329227995784916e+36f,
+    2.658455991569832e+36f,     5.316911983139664e+36f,     1.0633823966279327e+37f,    2.1267647932558654e+37f,
+    4.253529586511731e+37f,     8.507059173023462e+37f,     _2_power_127,               f32_nan,
+};
+// clang-format on
+
+using f8m8e0_params = std::tuple<uint8_t, float>;
+class F8E8M0PTest : public testing::TestWithParam<f8m8e0_params> {};
+
+INSTANTIATE_TEST_SUITE_P(convert,
+                         F8E8M0PTest,
+                         testing::ValuesIn(enumerate(exp_floats)),
+                         testing::PrintToStringParamName());
+
+TEST_P(F8E8M0PTest, f8_bits_to_f32) {
+    const auto& params = GetParam();
+    const auto& exp_value = std::get<1>(params);
+    const auto f8 = ov::float8_e8m0::from_bits(std::get<0>(params));
+
+    if (std::isnan(exp_value)) {
+        EXPECT_TRUE(std::isnan(static_cast<float>(f8)));
+    } else {
+        EXPECT_EQ(static_cast<float>(f8), exp_value);
+    }
+}
+
+TEST_P(F8E8M0PTest, f32_to_f8_bits) {
+    const auto& params = GetParam();
+    const auto& value = std::get<1>(params);
+    auto exp_value = std::get<0>(params);
+    const auto f8 = ov::float8_e8m0(value);
+    const auto f8_bits = f8.to_bits();
+
+    if (exp_value == 0xFF) {
+        EXPECT_TRUE(std::isnan(f8));
+    } else {
+        EXPECT_EQ(f8_bits, exp_value);
+    }
+}
+
+// clang-format off
+std::vector<std::tuple<float, float>> f32_f8_f32{
```

Comment:
```suggestion
const auto test_values = testing::Values<f8m8e0_params_cast>({}, {}, ...);
```

Rename `f32_f8_f32` to some meaningful name. Current name suggests three values but there are only two floats.

Reply:
done